### PR TITLE
hack,charts,manifests: Update the manual OLM metering manifests.

### PR DESCRIPTION
**Pull request by timflannagan1**
This updates our manual olm manifests for deploying Metering. The CatalogSourceConfig has been a depreciated object for a while now, so that's been replaced with using a ConfigMap-based CatalogSource implementation. Note: there's currently no ConfigMap that gets templated, but there is a `./hack/create-upgrade-configmap.sh` script that achieves the same functionality that can be renamed/generalized in down the line.

### CODE REVIEW

File: hack/olm-install.sh [hunk 1]
```diff
@@ -12,37 +12,41 @@ if [ "$METERING_NAMESPACE" != "openshift-metering" ]; then
...
         > "$TMPDIR/metering.subscription.yaml"
 
-        export OLM_MANIFESTS_DIR="$TMPDIR"
+    export OLM_MANIFESTS_DIR="$TMPDIR"
+    export NAMESPACE=$METERING_NAMESPACE
 fi
 
-msg "Installing Metering Catalog Source Config"
+msg "Creating the Metering ConfigMap"
+"$ROOT_DIR/hack/create-upgrade-configmap.sh" \
```
Comment (pweil-) [inline hack/olm-install.sh:36]: trailing `\` not necessary here?

File: pkg/deploy/uninstall.go [hunk 2]
```diff
@@ -74,7 +74,6 @@ func (deploy *Deployer) uninstallMeteringCSV() error {
 	// and hope that the user is re-running the olm-uninstall command.
 	sub, err := deploy.olmV1Alpha1Client.Subscriptions(deploy.config.Namespace).Get(deploy.config.SubscriptionName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		deploy.logger.Warnf("The metering subscription does not exist")
```
Comment (bentito) [inline pkg/deploy/uninstall.go:77]: Why are we still bothering with the conditional?

Comment (pweil-) [reply pkg/deploy/uninstall.go:77]: Not found is usually an acceptable condition during an uninstall since the intent is to remove the object.

Comment (bentito) [reply pkg/deploy/uninstall.go:77]: Aj, so the `IsNotFound` is trapping out the uninstall case. Should add a comment there to that effect then maybe.

Comment (pweil-) [reply pkg/deploy/uninstall.go:77]:
```go
// in the case where the subscription resource does not already exist, exit early
// and hope that the user is re-running the olm-uninstall command.
```

maybe not the clearest, but I think that's the intent 👍

**Comment by timflannagan1** [comment]
Overriding valid-bz label - this is just for QE testing.

**Pull request merged**